Vic2ToHoI4/Source/V2World/Technology/Inventions.cpp

```diff
@@ -0,0 +1,17 @@
+#include "Inventions.h"
+#include "Log.h"
+#include "OSCompatibilityLayer.h"
+#include "ParserHelpers.h"
+
+
+
+std::optional<std::string> Vic2::Inventions::getInventionName(int inventionNumber) const
+{
+	if (static_cast<size_t>(inventionNumber) > inventionNames.size())
+	{
+		Log(LogLevel::Warning) << "Invalid invention. Is this using a mod that changed inventions?";
+		return {};
+	}
+
+	return inventionNames[static_cast<size_t>(inventionNumber) - 1];
```

Comment:
static_cast<size_t>(inventionNumber - 1). You can't subtract from size_t without a potential wraparound while casting a  negative will straight return an error.

Reply:
(I think).